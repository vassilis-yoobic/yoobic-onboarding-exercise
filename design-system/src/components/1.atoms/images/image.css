/* No CSS *//*# sourceMappingURL=image.css.map */