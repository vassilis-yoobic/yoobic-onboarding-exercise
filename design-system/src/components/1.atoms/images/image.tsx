import { Component, ComponentInterface, h, Host, Prop } from '@stencil/core';

import { Identifier } from '../../shared/identifier';

@Component({
  tag: 'st-image',
  styleUrl: 'image.css',
})

export class Image implements ComponentInterface {
  @Prop() identifier?: Identifier = 'img';
  @Prop() src?: string;
  @Prop() alt?: string;
  @Prop() size: "Small" | "Medium" | "Large";

  render() {
    return (
      <Host>
        <img class={`img ${this.size}`} src={`${this.src}`} alt={`${this.alt}`} />
      </Host>);
  }
}
