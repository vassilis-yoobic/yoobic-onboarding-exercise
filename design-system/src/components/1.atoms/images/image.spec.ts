import { newSpecPage } from "@stencil/core/testing";

import { Image } from './image'

describe('1.atoms/images', () => {
  describe('rendering', () => {
    let page;

    beforeEach(async () => {
      page = await newSpecPage({
        components: [Image],
        html: '<st-image />'
      })
    })
    describe('props', () => {
      it('should render', () => {
        expect(page.root).toMatchSnapshot();
      })
      it('should render with small size', async () => {
        page.root.size = 'small';
        await page.waitForChanges();
        expect(page.root).toMatchSnapshot();
      })
    })
  })
})
