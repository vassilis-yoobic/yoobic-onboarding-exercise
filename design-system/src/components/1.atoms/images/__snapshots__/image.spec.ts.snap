// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1.atoms/images rendering props should render 1`] = `
<st-image>
  <img alt="undefined" class="img undefined" src="undefined">
</st-image>
`;

exports[`1.atoms/images rendering props should render with small size 1`] = `
<st-image>
  <img alt="undefined" class="img small" src="undefined">
</st-image>
`;
